pkg/client/cache/listers.go

```diff
@@ -225,6 +225,59 @@ func (s *StoreToReplicationControllerLister) GetPodControllers(pod *api.Pod) (co
 	return
 }
 
+// StoreToDaemonLister gives a store List and Exists methods. The store must contain only Daemons.
+type StoreToDaemonLister struct {
+	Store
+}
+
+// Exists checks if the given dc exists in the store.
+func (s *StoreToDaemonLister) Exists(daemon *api.Daemon) (bool, error) {
+	_, exists, err := s.Store.Get(daemon)
+	if err != nil {
+		return false, err
+	}
+	return exists, nil
+}
+
+// StoreToDaemonLister lists all daemons in the store.
+// TODO: converge on the interface in pkg/client
+func (s *StoreToDaemonLister) List() (daemons []api.Daemon, err error) {
+	for _, c := range s.Store.List() {
+		daemons = append(daemons, *(c.(*api.Daemon)))
+	}
+	return daemons, nil
+}
+
+// GetPodDaemon returns a list of daemon daemons managing a pod. Returns an error iff no matching daemons are found.
+func (s *StoreToDaemonLister) GetPodDaemon(pod *api.Pod) (daemons []api.Daemon, err error) {
+	var selector labels.Selector
+	var dc api.Daemon
+
+	if len(pod.Labels) == 0 {
+		err = fmt.Errorf("No daemons found for pod %v because it has no labels", pod.Name)
+		return
+	}
+
+	for _, m := range s.Store.List() {
+		dc = *m.(*api.Daemon)
+		if dc.Namespace != pod.Namespace {
+			continue
+		}
+		labelSet := labels.Set(dc.Spec.Selector)
+		selector = labels.Set(dc.Spec.Selector).AsSelector()
+
+		// If an dc with a nil or empty selector creeps in, it should match nothing, not everything.
+		if labelSet.AsSelector().Empty() || !selector.Matches(labels.Set(pod.Labels)) {
```

Comment:
Do you need both labelSet and selector? It seems like you're really only using selector (`labelSet.AsSelector().Empty()` -> `selector.Empty()`)